app.rb

```diff
@@ -1,8 +1,79 @@
 require 'sinatra/base'
+require './lib/rps_game'
+require './lib/player'
 class RockPaperScissors < Sinatra::Base
-  get '/test' do
-    'test page'
+  enable :sessions
+
+  get '/' do
+    session[:multiplayer] = false
+    erb :index
+  end
+
+  post '/name' do 
+    session[:player_1_name] = params[:player_1_name]
+    redirect '/one_player_game'
+  end
+
+  get '/one_player_game' do
+    @name = session[:player_1_name]
+    erb :game
+  end
+
+  post '/choice' do
```

Comment:
Perhaps there's a way of using this controller action for both single and multiplayer games.  You could, for example, use `session[:multiplayer]` to figure out whether you need to re-render the weapon choice form, or move on to the result page.

Alternatively, you could have a Game class that 'knows' whether or not another player's input is needed.

For the first option you'd need something in your controller / session that tracks the 'current turn' so that you can assign values in the session correctly.  For the second option, the Game would probably 'know' this as well.  E.g.

class Game

def choice(weapon)
  if @player1_weapon.nil?
    @player1_weapon = weapon
  else
    @player2_weapon = weapon
  end
end

def pending_weapon_choice?
  if multiplayer?
    !!@player1_weapon && !!@player2_weapon
  else
    !!@player1_weapon
  end
end
end
```

Then, inside this controller action...

```
game = session[:game]
if game.pending_weapon_choice?
  @current_player = game.player2
  erb :game 
else
  # the one player stuff
end
```

I've missed out bits of code for brevity but if that has made things unclear I can add some more context!

Reply:
If you like this option, you could use it for names as well.  Or you could combine weapon choice and name input into the same form